### DISCUSSION TITLE
530 From/Sender name is not valid with Telnet

### DISCUSSION THREAD

**Discussion by eriktroost070** [postalserver/postal#1953] [Help with using Postal]
We have a Postal relay setup for several of our services, we now want to test a connection from our other datacenter to Postal and use it as a relay.

I want to show our developers confirmation that sending is possible form the cluster (Kubernetes cluster on that other datacenter), and try to do this by using a Ubuntu container and Telnet (seems te easiest way to go for a small test).

I'm getting constant 530 when testing with Telnet, but I'm confident I'm using the correct domain. 
Is this some incompatibility with Postal and Telnet?

Here is my domain:
![Screenshot 2022-04-12 at 14 38 52](https://user-images.githubusercontent.com/74655015/162965037-e13c872f-5930-40f1-a017-8bb9c12da55d.png)

Here is the Telnet response I get:  
<img width="343" alt="Screenshot 2022-04-12 at 14 39 36" src="https://user-images.githubusercontent.com/74655015/162965434-b997d40d-4a8a-4432-b0c0-3aeb29428335.png">

As you can see authentication is successful (which indicates for me the correct credentials are used). Tried several of the other mailservers we have configured in Postal (all the same domain) but no luck. These mailservers do however work when sending from an application to the relay. Just not from Telnet. Which makes it hard to convince the devs the connection is working from the Kubernetes cluster.

**Comment by willpower232** [comment]
As you say everything matches so it must be that the credential you are using is not in the same organization

**Reply by kayastha1988** [reply]
**"it must be that the credential you are using in organization"**
Would you plz elaborate this a bit please? Where I can get this information? Looks like I have tried with same credentials but still I have getting same error.
looking forward your help.
thanks

**Comment by eriktroost070** [comment]
We only have 1 organisation and 1 domain we use kpnthings.com
We do have several mail servers setup in Postal te separate those. 

And why can I authenticate if the credentials are not correct? Because if I put something random in the credentials it give a 535 invalid credentials.

![Screenshot 2022-04-12 at 16 31 20](https://user-images.githubusercontent.com/74655015/162986582-0601c0d9-bb3c-41b1-8418-c1378fa61708.png)

![Screenshot 2022-04-12 at 16 31 58](https://user-images.githubusercontent.com/74655015/162986602-02fab534-ec6e-4185-af2c-bfe724e45c57.png)

**Reply by willpower232** [reply]
the credentials are correct, the problem is that kpnthings.com isn't apparently in the list of domains within that server.

**Reply by eriktroost070** [reply]
We only have 1 domain on Postal, and that is kpnthings.com. Tried different credentials (we have multiple mail servers in Postal), all of which have kpnthings.com as a domain. But it did not matter, with all credentials I get a 503.

Sending from an application to those mail servers is not an issue, with the credentials that get a 503 with Telnet works like a charm on those other applications (they use postfix underneath I think). It seems to be only when I try a test with Telnet.

Anyway, not really a show stopper, but was a bit disappointed that I was not able to do a simple test of our new network route and see if it works or not. I guess it works, because we can make a connection.

**Reply by willpower232** [reply]
Odd indeed, you can see that error message only comes from one place, where it can't match the domain

https://github.com/postalserver/postal/blob/9f0697f194209f5fae5e451ba8fb888413fe37fa/lib/postal/smtp_server/client.rb#L434
